rmw_fastrtps_shared_cpp/src/rmw_node.cpp

```diff
@@ -42,45 +47,60 @@ __rmw_create_node(
   const char * name,
   const char * namespace_)
 {
-  if (!name) {
-    RMW_SET_ERROR_MSG("name is null");
+  assert(nullptr != context);
+  assert(nullptr != context->impl);
+  assert(identifier == context->implementation_identifier);
+  static_cast<void>(identifier);
+
+  int validation_result = -1;
```

Comment:
nit: maybe defaulting to RMW_NODE_NAME_INVALID?
there's a unlikely chance of RMW_NODE_NAME_VALID being `-1`

Reply:
I took a quick look at the uses of `rmw_validate_node_name` throughout the code base.  Currently, how we initialize `validation_result` is inconsistent.  Some places, we don't initialize it at all, and the caller assumes `rmw_validate_node_name` fills it in with something valid.  Some places, we initialize it to 0.  And it also looks like the values that it can possibly return are just `#define`d: https://github.com/ros2/rmw/blob/ab3df8095cedf2c809c631e5c3442f964404e797/rmw/include/rmw/validate_node_name.h#L26 .

What I'll suggest here is:

1.  Switch this to `RMW_NODE_NAME_VALID` (since there is no generic `RMW_NODE_NAME_INVALID`)
1.  (separate PR) Change the definition of these values to an enum in `rmw/validate_node_name.h`
1.  (separate PR) Add an enum value of `RMW_NODE_NAME_UNKNOWN`.
1.  (separate PR) Update all of the callers of this method to use that.

What do people think of that plan?

Reply:
> (separate PR) Add an enum value of RMW_NODE_NAME_UNKNOWN.

This new value would be for initializing the enum, right?

> What do people think of that plan?

:+1:

Reply:
> This new value would be for initializing the enum, right?

Yeah, exactly.  It would ensure we never end up in a UB situation.

Reply:
Ok, I'll start out with option (1.) and then proceed with the rest in follow-up PRs.